src/permissions_template.c

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+
+#include "kerneltypes.h"
+#include "kmalloc.h"
+#include "permissions_template.h"
+#include "console.h"
+
+#define MAX_NUMBER_OF_PERMISSIONS 169 // temporary, for now they'll be statically allocated
+
+static uint32_t template_counter = 1;
+static struct permissions_template* permissions_table[MAX_NUMBER_OF_PERMISSIONS]; // statically allocated array of templates
+
+uint32_t generate_new_identifier() {
+    return template_counter++;
+}
+
+struct process_permissions* permissions_from_template(struct permissions_template *template) {
+
+    if (template <= 0) {
+        console_printf("Error: invalid template identifier\n");
```

Comment:
Not really a correct error message here? Its the template that invalid, not necessarily the identifier